### DISCUSSION TITLE
Best way to save data to database

### DISCUSSION THREAD

**Discussion by ChrisSAV-scal** [pymeasure/pymeasure#682] [Q&A]
Hi, 
What is the best way for saving the measurements  in my database ?  Do I need to create a new thread or I can to use the same thread that csv file ?
Can you help me with an example ?
Thanks in advance.

**Comment by neal-kepler** [comment]
I think it depends on how long the call to save the measurements to the db takes, what data you are trying to save, and how much time you have.

If the call is fast (or you don't mind the performance penalty), you don't want to write a new handler and you want to save the data as the csv's you could do it after the call to emit in the execute method of the procedure. I would also create the db connection in the startup method and the close the db connection in the shutdown method

If the call is slow, and/or you have the time, I think the "right" or best way to do it would be to write a new handler. #277 implements a handler that allows dumping from a dictionary or pandas dataframe directly, and shows how to modify the handler, but this is an ongoing problem (see #136). Reading and writing to a database would also require redoing how to parse the header info with all the experimental params.

What data do you want to save? Do you want the header as like a json blob in the same db? Are you using sqlalchemy to connect to your db?
